dcap/types/src/quote3.rs

```diff
@@ -4,10 +4,43 @@
 
 #[cfg(feature = "alloc")]
 use alloc::vec::Vec;
+use core::mem;
 use mc_sgx_core_types::{QuoteNonce, ReportData};
+use mc_sgx_dcap_sys_types::{sgx_ql_ecdsa_sig_data_t, sgx_quote3_t};
 use sha2::{Digest, Sha256};
 use subtle::ConstantTimeEq;
 
+// The size of the quote bytes. Not including the authentication or
+// certification data.
+const QUOTE_SIZE: usize =
+    mem::size_of::<sgx_quote3_t>() + mem::size_of::<sgx_ql_ecdsa_sig_data_t>();
+
+// The offset to the authentication data
+const AUTH_DATA_OFFSET: usize = QUOTE_SIZE;
+
+/// The minimum size of a byte array to contain a [`Quote3`]
+///
+// 8 is from the 2 bytes for QE authentication data size and 2(type) + 4(size)
+// for QE certification data
+pub const MIN_QUOTE_SIZE: usize = QUOTE_SIZE + 8;
+
+/// Errors interacting with a Quote3
+#[derive(Clone, Debug, displaydoc::Display, Eq, Hash, PartialEq, PartialOrd, Ord)]
+#[non_exhaustive]
+pub enum Error {
+    /** Quote buffer too small; actual size: {actual_size}, required size
+     * {required_size} */
+    #[allow(missing_docs)]
+    InvalidInputLength {
```

Comment:
Thoughts on `Invalid` in the name.
It's already an error so perhaps `Invalid` is unecessary emphasis

Reply:
Removed invalid